src/main/java/io/debezium/connector/jdbc/dialect/sqlserver/SqlServerDatabaseDialect.java

```diff
@@ -47,6 +47,27 @@ private SqlServerDatabaseDialect(JdbcSinkConnectorConfig config, SessionFactory
         super(config, sessionFactory);
     }
 
+    @Override
+    public String getInsertStatement(TableDescriptor table, SinkRecordDescriptor record) {
+        String insertStatement = super.getInsertStatement(table, record);
+        return wrapWithIdentityInsert(table, insertStatement);
+    }
+
+    private String wrapWithIdentityInsert(TableDescriptor table, String sqlStatement) {
+        boolean contains = getConfig().getSqlServerIdentityTableNames().contains(table.getId().getTableName());
+        if (!contains) {
+            return sqlStatement;
+        }
```

Comment:
As I mentioned in my comment on the format of this field, we should be comparing the schema+table names if we use a property configuration that specifies names.

However, I am curious what this check's performance impact is compared to other alternatives.  For example, what if instead we checked if a column was an auto-generated identity column while creating the `TableDescriptor` instance, in which we could then expose a `hasAutoGeneratedIdentityColumn()` method.  If so, then here it would be a simple boolean check like:

```
if (table.hasAutoGeneratedIdentityColumn() && !getConfig().isSqlServerIdentityInserts()) {
  return sqlStatement;
}
// generate the wrapping statements here
```

wdyt?